src/modules/partition/jobs/CreatePartitionJob.cpp

```diff
@@ -32,9 +33,134 @@ CreatePartitionJob::CreatePartitionJob( Device* device, Partition* partition )
+    { "4301d2a6-4e3b-4b2a-bb94-9e0b2c4225ea", "/usr Partition (Itanium/IA-64)" },
+    { "8f461b0d-14ee-4e81-9aa9-049b6fb97abd", "/usr Verity Partition (x86)" },
+    { "77ff5f63-e7b6-4633-acf4-1565b864c0e6", "/usr Verity Partition (x86-64)" },
+    { "c215d751-7bcd-4649-be90-6627490a4c05", "/usr Verity Partition (32-bit ARM)" },
+    { "6e11a4e7-fbca-4ded-b9e9-e1a512bb664e", "/usr Verity Partition (64-bit ARM/AArch64)" },
+    { "6a491e03-3be7-4545-8e38-83320e0ea880", "/usr Verity Partition (Itanium/IA-64)" },
+    { "933ac7e1-2eb4-4f13-b844-0e14e2aef915", "Home Partition" },
+    { "3b8f8425-20e0-4f3b-907f-1a25a76f98e8", "Server Data Partition" },
+    { "4d21b016-b534-45c2-a9fb-5c16e091fd2d", "Variable Data Partition" },
+    { "7ec6f557-3bc5-4aca-b293-16ef5df639d1", "Temporary Data Partition" },
+    { "0657fd6d-a4ab-43c4-84e5-0933c84b4f4f", "Swap" },
+    { "c12a7328-f81f-11d2-ba4b-00a0c93ec93b", "EFI System Partition" },
+    { "bc13c2ff-59e6-4262-a352-b275fd6f7172", "Extended Boot Loader Partition" },
+    { "0fc63daf-8483-4772-8e79-3d69d8477de4", "Other Data Partitions" },
+    { "ebd0a0a2-b9e5-4433-87c0-68b6b72699c7", "Microsoft basic data" },
+};
+
+static QString
+prettyGptType( const QString& type )
+{
+    return gptTypePrettyStrings.value( type.toLower(), type );
+}
+
+static QString
+prettyGptAttributes( quint64 attributes )
+{
+    QStringList list;
+
+    if ( attributes & ( 1UL << 0 ) )
+    {
+        list += QString( "RequiredPartition" );
+    }
+
+    if ( attributes & ( 1UL << 1 ) )
+    {
+        list += QString( "NoBlockIOProtocol" );
+    }
+
+    if ( attributes & ( 1UL << 2 ) )
+    {
+        list += QString( "LegacyBIOSBootable" );
+    }
+
+    for ( qint8 bit = 3; bit < 64; bit++ )
+    {
+        if ( attributes & ( 1UL << bit ) )
+        {
+            list += QString( "Bit" ) + QString::number( bit );
+        }
+    }
+
+    QString string = list.join( QStringLiteral( " and " ) );
+    if ( string.size() > 64 )   
+    {
+        return QString( "0x" ) + QString::number( attributes, 0x10 );
+    }
+
+    return string;
+}
+
+static QString
+prettyGptEntries( const Partition* partition )
+{
+    if ( !partition )
+    {
+        return QString("");
+    }
+
+    QStringList list;
+
+    if ( partition->label().size() )
+    {
+        list += partition->label();
+    }
+
+    if ( partition->uuid().size() )
+    {
+        list += partition->uuid();
+    }
+
+    QString type = prettyGptType( partition->type() );
+    if ( type.size() )
+    {
+        list += type;
+    }
+
+    QString attributes = prettyGptAttributes( partition->attributes() );
+    if ( attributes.size() && attributes != QString( "0" ) )
+    {
+        list += attributes;
+    }
+
+    return list.join( QStringLiteral( ", " ) );
+}
+
 QString
 CreatePartitionJob::prettyName() const
 {
+    if ( CalamaresUtils::Partition::getPartitionTable( m_partition )->type() == PartitionTable::TableType::gpt )
```

Comment:
The function can return nullptr (it's a screwed up system then, but still).

Reply:
Good catch